universe/base.go

```diff
@@ -165,7 +165,46 @@ func (a *MintingArchive) RegisterIssuance(ctx context.Context, id Identifier,
 	// it as a file first as that's what the expected wants.
 	//
 	// TODO(roasbeef): add option to skip proof verification?
-	assetSnapshot, err := leaf.GenesisProof.Verify(
+	assetSnapshot, err := a.verifyIssuanceProof(ctx, &IssuanceItem{
+		ID:   id,
+		Key:  key,
+		Leaf: leaf,
```

Comment:
Meta reveal never actually set in the struct?

Reply:
Yeah, I guess in this case it doesn't make sense to use the `IssuanceItem` struct, since the meta reveal is only known _after_ verifying the proof. Reverted to individual parameters.